src/hooks/useTodoList.tsx

```diff
@@ -0,0 +1,88 @@
+import { useEffect, useRef, useState } from 'react';
+import { addTodo, deleteTodo, getTodos } from '../api/todos';
+import { Todo } from '../types/Todo';
+
+export const useTodoList = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [errorMessage, setErrorMessage] = useState<string | null>(null);
```

Comment:
```suggestion
  const [errorMessage, setErrorMessage] = useState<string | null>(null);
```
use enum for error messages